enumerable_methods.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      x = []
+      i = 0
+      while i < arr.length
+        x.push(yield(arr[i]))
+        i += 1
+      end
+      x
+    else
+      puts 'Error: Parameter is not an array!'
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      x = []
+      i = 0
+      while i < arr.length
+        x.push(yield(arr[i], i))
+        i += 1
+      end
+      x
+    else
+      puts 'Error: Parameter is not an array!'
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    x = []
+    i = 0
+    arr = self
+    if arr.is_a? Array
+      while i < arr.length
+        x.push(arr[i]) if yield(arr[i])
+        i += 1
+      end
+      x
+    else
+      puts 'Error: Parameter is not an array!'
+    end
+  end
+
+  def my_all?
+    return to_enum unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      x = true
+      i = 0
+      while i < arr.length
+        x = false unless yield(arr[i])
+        i += 1
+      end
+      x
+    else
+      puts 'Error: Parameter is not an array!'
+    end
+  end
+
+  def my_any?
+    return to_enum unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      i = 0
+      while i < arr.length
+        return true if yield(arr[i])
+
+        i += 1
+      end
+      false
+    else
+      puts 'Error: Parameter is not an array!'
+    end
+  end
```

Comment:
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern